http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -1141,9 +1141,10 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
     samVisibleAppsOpt <- NonEmptyList.fromList(samResources).traverse { apps =>
       authProvider.filterUserVisible(apps, userInfo)
     }
-
     res = samVisibleAppsOpt match {
-      case None => Vector.empty
+      case None =>
+        authProvider.checkUserEnabled(userInfo)
```

Comment:
So this is a common Scala pitfall. The `authProvider.checkUserEnabled` won't actually execute here because the `F` it returns is not chained to the surrounding for-comp (flatMap). 

In a for-comp, `<-` should be used when the RHS expression is an `F[..]` and `=` should be used when the RHS is a pure expression.

So this will execute `f1`, `f2`, `f3` in sequence:
```
for {
  _ <- f1
  _ <- f2
  x = 42
  _ <- f3(x)
} yield ()
```

This is wrong:
```
for {
  _ <- f1
  _ = f2 // will not execute!!
  _ <- f3
} yield ()
```

So I think this code needs to look something like:
```
res <- samVisibleAppsOpt match {
  case None => 
    // returns F[Vector[something]]
    authProvider.checkUserEnabled(userInfo) >> Vector.empty
  case Some(samVisibleApps) =>
    // same logic but now needs to return an F[Vector[something]] instead of Vector[something]
}
```

Reply:
Btw I know it was my idea -- but if changing the `match` expression is too unwieldy in practice could also go back to separate check:
```
_ <- if (samVisibleAppsOpt.isEmpty)  authProvider.checkUserEnabled(userInfo) else F.unit 
_ <- samVisibleAppsOpt match { … }
```
(I don't really care that much, so whatever is more readable :))

Reply:
Thank you!! Question, would I be able to accomplish the same thing with a `getOrElse`?
```
    samVisibleAppsOpt <- NonEmptyList.fromList(samResources).traverse { apps =>
      authProvider.filterUserVisible(apps, userInfo)
    }.getOrElse(authProvider.checkUserEnabled(userInfo))
```

Reply:
Oh, no, that wouldn't work, because then it wouldn't be a None...

Reply:
What if I snuck it into its own line? Then I wouldn't have to pollute the cases with a new type wrapper:
```
    _ <- samVisibleAppsOpt.getOrElse(authProvider.checkUserEnabled(userInfo))
```

Reply:
Maybe I should stop liveblogging my ignorance ^-^

Reply:
```
  _ <- samVisibleAppsOpt.getOrElse(authProvider.checkUserEnabled(userInfo))
```
The types don't work here. The RHS needs to return a `F[Unit]`. In the above code the `else` case returns an `F[Unit]` but not the `get` case. I didn't try but it probably doesn't compile..

Reply:
Yeah, I had realized :) I'll get there

Reply:
This would type check:
```
 _ <- samVisibleAppsOpt.map(_ => F.unit).getOrElse(authProvider.checkUserEnabled(userInfo))
```

Reply:
Type checking is suggesting that the Option here is wrapping a List, not a NonEmptyList anymore (maybe it lost that type in the traverse step?)

So I might need to cover the case of Some([]) | None after all.

Reply:
I moved the checks to the end of the methods, since many of them were filtering inside the cases and I wanted to keep this access check a little bit separate for clarity.